src/main/java/org/myrobotlab/codec/CodecUtils.java

```diff
@@ -270,20 +275,50 @@ public static String capitalize(final String line) {
      * using the selected JSON backend.
      *
      * @param json  The JSON to be deserialized in String form
-     * @param clazz The target class.
+     * @param clazz The target class. If a class is not supplied the default class returned will be an LinkedHashMap
      * @param <T>   The type of the target class.
      * @return An object of the specified class (or a subclass of) with the state
      * given by the json. Null is an allowed return object.
      * @throws JsonDeserializationException if an error during deserialization occurs.
      * @see #USING_GSON
      */
+    @SuppressWarnings("unchecked")
     public static <T> /*@Nullable*/ T fromJson(/*@Nonnull*/ String json, /*@Nonnull*/ Class<T> clazz) {
         try {
             if (USING_GSON) {
+                if (clazz == null) {
+                  clazz = (Class<T>)LinkedHashMap.class;
+                }
                 return gson.fromJson(json, clazz);
+            } else {
+              if (clazz == null) {
+ 
+                JsonFactory factory = new JsonFactory();
+                JsonParser parser = factory.createParser(json);

```

Comment:
If we keep this code this should be wrapped in a try-with-resources block so the parser is auto-closed

Reply:
so ...

```java
                JsonParser parser = mapper.getFactory().createParser(json);
                
                // "peek" at the next token to determine its type
                JsonToken token = parser.nextToken();

                if (token == JsonToken.START_OBJECT) {
                  clazz = (Class<T>)Map.class;
                } else if (token == JsonToken.START_ARRAY) {
                  clazz = (Class<T>)ArrayList.class;
                } else if (token.isScalarValue()) {
                  JsonNode node = mapper.readTree(json);
                  if (node.isInt()) {
                      return mapper.readValue(json, (Class<T>)Integer.class);
                  } else if (node.isDouble()) {
                      return mapper.readValue(json, (Class<T>)Double.class);
                  } else if (node.isTextual()) {
                    return mapper.readValue(json, (Class<T>)String.class);
                  }
                } else {
                    log.error("could not derive type from peeking json {}", json);
                }
                
                parser.close();

```                
              the object factory is used from the static mapper - but i think the close() could be lethal

Reply:
I previously added this to the unit tests 
```java

    // primitives objects arrays
    Object test = CodecUtils.fromJson("1");
    assertEquals(test.getClass(), Integer.class);

    test = CodecUtils.fromJson("1.0");
    assertEquals(test.getClass(), Double.class);
    
    test = CodecUtils.fromJson("\"Ahoy!\"");
    assertEquals(test.getClass(), String.class);
    
    test = CodecUtils.fromJson("[]");
    assertEquals(test.getClass(), ArrayList.class);
    
    test = CodecUtils.fromJson("{}");
    assertEquals(test.getClass(), LinkedHashMap.class);
``` 
and it seems worky

Reply:
factory needs to be closed ? would you mind doing a pr for that ?